options/posix/generic/pthread-stubs.cpp

```diff
@@ -499,9 +499,15 @@ int pthread_getname_np(pthread_t, char *, size_t) {
 	__builtin_unreachable();
 }
 
-int pthread_setschedparam(pthread_t, int, const struct sched_param *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int pthread_setschedparam(pthread_t thread, int policy, const struct sched_param *param) {
+	auto tcb = reinterpret_cast<Tcb*>(thread);
+
+	MLIBC_CHECK_OR_ENOSYS(mlibc::sys_setschedparam, ENOSYS);
```

Comment:
wrong return value? (this isn't necessarily against spec afaik, but I don't think we should give return a mlibc-specific meaning of the return value)

Reply:
> On success, these functions return 0; on error, they return a nonzero error number.

I don't see how we could do any different TBH.

Reply:
-1?

Reply:
oh, error number, I misread the manpage, sorry